### PR TITLE
Fix escaped strings for TryGetDateTime(Offset)

### LINKED ISSUE #62720: JsonElement.TryGetDateTime throws IndexOutOfRangeException
### Description

In some cases `JsonElement.TryGetDateTime` throws an exception. It happens when value is string containing two '>' and having length 2 or 3 (for example, ">>", ">> a", "> a>", etc.).

### Reproduction Steps

```cs
string json = JsonSerializer.Serialize(new { X = ">>a"});		
JsonDocument jsonDocument = JsonDocument.Parse(json);
jsonDocument.RootElement.GetProperty("X").TryGetDateTime(out var x); //IndexOutOfRangeException
```

### Expected behavior

return false from method

### Actual behavior

**IndexOutOfRangeException**
at System.Text.Json.JsonHelpers.TryParseDateTimeOffset(ReadOnlySpan`1 source, DateTimeParseData& parseData)
at System.Text.Json.JsonHelpers.TryParseAsISO(ReadOnlySpan`1 source, DateTime& value)
at System.Text.Json.JsonReaderHelper.TryGetEscapedDateTime(ReadOnlySpan`1 source, DateTime& value)
at System.Text.Json.JsonDocument.TryGetValue(Int32 index, DateTime& value)
at System.Text.Json.JsonElement.TryGetDateTime(DateTime& value)

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by WinCPP**
Fix the exception thrown when the escaped DateTime(Offset)
property value is unescaped and it becomes shorter than
the minimum expected length. In such a case, the Try* method
should fail and return false.

Fixes #62720

### CODE REVIEW

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonReaderTests.cs [hunk 1]
```diff
@@ -4534,5 +4535,14 @@ public static IEnumerable<object[]> GetCommentUnescapeData
                 return dataList;
             }
         }
+
+        public static IEnumerable<object[]> InvalidDateTimePropertyValueData()
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonReaderTests.cs:4539]: Nit: shouldn't this helper method be collocated with the source file of the test?

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs [hunk 2]
```diff
@@ -2047,6 +2047,42 @@ void assertOdd(JsonElement elem)
...
+                // DateTime value length more than System.Text.Json.JsonConstants.MaximumEscapedDateTimeOffsetParseLength = 252
+                foreach (var count in new int[] { 9, 253 })
+                {
+                    yield return new object[] { string.Join("", Enumerable.Repeat(str, count)) };
+                }
+            }
+        }
+
+
+        [Theory]
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs:2064]: While your changes do address the bug as originally reported, these tests are not validating that this is the case: I tried running them without your fix and they're still passing. Please add tests verifying that bug as originally reporting has been fixed.

**Review by eiriktsarpalis — CHANGES_REQUESTED** [review]

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonReaderTests.cs (hunk 1 above)
Comment (WinCPP) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonReaderTests.cs:4539]: Removed this method as I have used "InlineData" attribute now (more in the discussion below).

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs (hunk 2 above)
Comment (WinCPP) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs:2064]: My bad. I just kept thinking about the fact that the DateTime length should be between 10 and 252 and missed the real test case. I have added it now and removed the <10 and >252 test cases as they anyways succeed in any case.

**Comment by WinCPP** [comment]
@eiriktsarpalis please review when you get a moment. Also, I am contributing after a long time and I am forgetting how to re-trigger failed runs/checks. Kindly advise.

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs [hunk 3]
```diff
@@ -2047,6 +2047,29 @@ void assertOdd(JsonElement elem)
             }
         }
 
+        [Theory]
+        [InlineData(">>")]
+        [InlineData(">>>")]
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs:2052]: For completeness, could you also add some of the inputs as listed in the original report?

">>", ">> a", "> a>"

**Review by eiriktsarpalis — APPROVED** [review]
Other than a minor comment, LGTM. Thanks!

**Comment by eiriktsarpalis** [comment]
> Also, I am contributing after a long time and I am forgetting how to re-trigger failed runs/checks. Kindly advise.

I'll deal with the CI kinks. Thanks!

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs (hunk 3 above)
Comment (WinCPP) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs:2052]: Done!

**Pull request merged**